.github/workflows/integrate.yaml

```
@@ -0,0 +1,61 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  lint:
+    name: Lint Code
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        charm: [operator, serving, eventing]
+    steps:
+      - uses: actions/checkout@v2
+      - run: sudo apt update && sudo apt install tox
+      - run: tox -e ${{ matrix.charm }}-lint
+
+  unit:
+    name: Unit Test
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        charm: [operator, serving, eventing]
+    steps:
+      - uses: actions/checkout@v2
+      - run: sudo apt update && sudo apt install tox
+      - run: tox -e ${{ matrix.charm }}-unit
+
+  integration:
+    name: Integration Test
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@v2
+
+      - name: Setup operator environment
+        uses: charmed-kubernetes/actions-operator@main
+        with:
+          provider: microk8s
+          channel: 1.21/stable
+
+      - run: sudo snap install juju-wait --classic
+      - run: juju destroy-model --yes --destroy-storage testing
+      - run: juju add-model knative-test
+
+      - name: Run integration tests
+        run: tox -vve integration -- --model knative-test --destructive-mode
+        env:
+          KUBECONFIG: /home/runner/.kube/config
+
+      - run: kubectl get all -A
+        if: failure()
+
+      - run: juju status
+        if: failure()
```

Add the log dump action? 

```suggestion
        if: failure()
        
      - name: Dump logs
        uses: canonical/charm-logdump-action@main
        if: failure()
        with:
          app: knative-operator
          model: knative-test
```

---

Mmmh, seems like this is per charm, is it okay if we run this individually for each charm?

---

Alright, leaving the workflow as it is until that issue gets solved.